plugins/UM3NetworkPrinting/src/Cloud/CloudOutputDevice.py

```diff
@@ -140,9 +140,17 @@ def clusterData(self, value: CloudClusterResponse) -> None:
 
     ## Checks whether the given network key is found in the cloud's host name
     def matchesNetworkKey(self, network_key: str) -> bool:
-        # A network key looks like "ultimakersystem-aabbccdd0011._ultimaker._tcp.local."
+        # Typically, a network key looks like "ultimakersystem-aabbccdd0011._ultimaker._tcp.local."
         # the host name should then be "ultimakersystem-aabbccdd0011"
-        return network_key.startswith(self.clusterData.host_name)
+        if network_key.startswith(self.clusterData.host_name):
+            return True
+        
+        # However, for manually added printers, the local IP address is used in lieu of a proper
+        # network key, so check for that as well
+        if network_key == self.clusterData.host_internal_ip:
```

Comment:
Maybe something like network_key.find(self.clusterData.host_internal_ip) to be more robust for cases like when the IP suddenly contains http:// in front of it or something.

Reply:
Fair point 👍